### ISSUE TITLE
Tests against dev version of git-annex on Ubuntu keep getting stuck

### ISSUE THREAD

**Issue by jwodder** [datalad/datalad#7370]
### What is the problem?

Starting on April 10, [datalad/git-annex](https://github.com/datalad/git-annex)'s daily tests of DataLad on Ubuntu (for master, maint, & release) against the latest dev version of git-annex have stalled partway through, ultimately being killed by GitHub for exceeding the six hour time limit.

In all the stalling runs I've checked, the last test that finished was `datalad/support/tests/test_annexrepo.py::test_AnnexRepo_is_under_annex`, and the next test after that is `datalad/support/tests/test_annexrepo.py::test_AnnexRepo_web_remote`, which is presumably the test that's stalling.

**Comment by yarikoptic** [comment]
change in behavior is to no longer include `"file"` when input doesn't match annexed file path (but still a file in reality):

```shell
❯ git annex info --json part*
fatal: Not a valid object name participants.tsv
{"command":"info","error-messages":[],"file":"participants.tsv","input":["participants.tsv"],"note":"not a directory or an annexed file or a treeish or a remote or a uuid","success":false}
❯ (source ~/git-annexes/10.20230407+git63-g3d1d77a1bb.env; !!)
(source ~/git-annexes/10.20230407+git63-g3d1d77a1bb.env; git annex info --json part*)
fatal: Not a valid object name participants.tsv
{"command":"info","note":"not a directory or an annexed file or a treeish or a remote or a uuid","success":false,"input":["participants.tsv"],"error-messages":[],"file":null}
info: 1 failed
```
note: stall happens only when running all tests in that file, not only the stalling one.

**Comment by christian-monch** [comment]
> change in behavior is to no longer include `"file"` when input doesn't match annexed file path (but still a file in reality):
> 
> ```shell
> ❯ git annex info --json part*
> fatal: Not a valid object name participants.tsv
> {"command":"info","error-messages":[],"file":"participants.tsv","input":["participants.tsv"],"note":"not a directory or an annexed file or a treeish or a remote or a uuid","success":false}
> ❯ (source ~/git-annexes/10.20230407+git63-g3d1d77a1bb.env; !!)
> (source ~/git-annexes/10.20230407+git63-g3d1d77a1bb.env; git annex info --json part*)
> fatal: Not a valid object name participants.tsv
> {"command":"info","note":"not a directory or an annexed file or a treeish or a remote or a uuid","success":false,"input":["participants.tsv"],"error-messages":[],"file":null}
> info: 1 failed
> ```
> [...]

Maybe I misunderstood your example, but there is a `"file"`-key in the result of git annex (10.20230407+git63-g3d1d77a1bb). The value is `null` though. So the issue is that the value is `null`, instead of `"participants.tsv"`?

**Comment by yarikoptic** [comment]
<details>
<summary>FWIW: that test_AnnexRepo_web_remote passes for me locally when ran in isolation :-/</summary> 

```shell
❯ ( source ~/git-annexes/10.20230407+git201-g5df89d58c7.env; git annex version 2>&1 | head -n1 ; python3 -m pytest -s -v -k test_AnnexRepo_web_remote )
git-annex version: 10.20230407+git201-g5df89d58c7-1~ndall+1
====================================================== test session starts ======================================================
platform linux -- Python 3.11.2, pytest-7.2.1, pluggy-1.0.0 -- /home/yoh/proj/datalad/datalad-maint/venvs/dev3/bin/python3
cachedir: .pytest_cache
rootdir: /home/yoh/proj/datalad/datalad-maint, configfile: tox.ini
plugins: xdist-3.1.0, fail-slow-0.3.0, cov-4.0.0
collected 1285 items / 1284 deselected / 1 selected                                                                             

datalad/support/tests/test_annexrepo.py::test_AnnexRepo_web_remote PASSEDVersions: annexremote=1.6.0 boto=2.49.0 cmd:7z=16.02 cmd:annex=10.20230407+git201-g5df89d58c7-1~ndall+1 cmd:bundled-git=2.30.2 cmd:git=2.30.2 cmd:ssh=9.2p1 cmd:system-git=2.39.2 cmd:system-ssh=9.2p1 datalad=0.18.3+26.gdd0819399 humanize=4.5.0 iso8601=1.1.0 keyring=23.13.1 keyrings.alt=UNKNOWN msgpack=1.0.4 patoolib=1.12 platformdirs=2.6.2 requests=2.28.2 tqdm=4.64.1
Obscure filename: str=b' |;&%b5{}\'"<>\xce\x94\xd0\x99\xd7\xa7\xd9\x85\xe0\xb9\x97\xe3\x81\x82 .datc ' repr=' |;&%b5{}\'"<>ΔЙקم๗あ .datc '
Encodings: default='utf-8' filesystem='utf-8' locale.prefered='UTF-8'
Environment: PATH='/home/yoh/git-annexes/10.20230407+git201-g5df89d58c7/usr/bin:/home/yoh/proj/datalad/datalad-maint/venvs/dev3/bin:/home/yoh/gocode/bin:/home/yoh/gocode/bin:/home/yoh/bin:/home/yoh/.local/bin:/usr/local/bin:/usr/bin:/bin:/usr/local/games:/usr/games:/sbin:/usr/sbin:/usr/local/sbin' LANG='en_US.UTF-8' GIT_PAGER='less --no-init --quit-if-one-screen' GIT_CONFIG_PARAMETERS="'init.defaultBranch=dl-test-branch' 'clone.defaultRemoteName=dl-test-remote'" PYTHON_KEYRING_BACKEND='keyrings.alt.file.PlaintextKeyring' GIT_ASKPASS='true'


======================================================= warnings summary ========================================================
venvs/dev3/lib/python3.11/site-packages/requests_ftp/ftp.py:9
  /home/yoh/proj/datalad/datalad-maint/venvs/dev3/lib/python3.11/site-packages/requests_ftp/ftp.py:9: DeprecationWarning: 'cgi' is deprecated and slated for removal in Python 3.13
    import cgi

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
========================================= 1 passed, 1284 deselected, 1 warning in 3.44s =========================================
```
</details>

**Comment by jwodder** [comment]
@yarikoptic The "stuck" runs were the ones which were killed by GitHub Actions for exceeding the 6 hour time limit.  These are marked in the job summary listing with a grey '!' in a stop sign, as opposed to the white X in a red circle used for normal failure; moreover, the subject of the daily status e-mail reports such runs as "INCOMPLETE".  Based on that, the last time there were any stuck runs was on April 30 ([link to run](https://github.com/datalad/git-annex/actions/runs/4847924157)).

**Comment by yarikoptic** [comment]
Great, thanks! How did you determine which tests got stuck?

**Comment by jwodder** [comment]
@yarikoptic As I stated in the top comment:

> In all the stalling runs I've checked, the last test that finished was `datalad/support/tests/test_annexrepo.py::test_AnnexRepo_is_under_annex`, and the next test after that is `datalad/support/tests/test_annexrepo.py::test_AnnexRepo_web_remote`, which is presumably the test that's stalling.

**Comment by yarikoptic** [comment]
gotcha -- I misread your comment under the influence of watching the logs from our `-n 2` run where it is not that easy to determine where it is stalled.

**Comment by yarikoptic** [comment]
as we no longer have them stuck, let's close this one

**Comment by yarikoptic** [comment]
but the moral/my wild guess -- likely something in our runner handling of errorneous execution doesn't quit the process properly

**Comment by yarikoptic-gitmate** [comment]
Issue fixed in [`0.19.1`](https://github.com/datalad/datalad/releases/tag/0.19.1)
